### ISSUE TITLE
Unstable integration test azure_rm_containerinstance

### ISSUE THREAD

**Issue by mattclay** [ansible/ansible#36080]
##### ISSUE TYPE

Bug Report

##### COMPONENT NAME

azure_rm_containerinstance integration test

##### ANSIBLE VERSION

2.5

##### CONFIGURATION

N/A

##### OS / ENVIRONMENT

Shippable

##### SUMMARY

Unstable integration test azure_rm_containerinstance:

https://app.shippable.com/github/ansible/ansible/runs/54332/51/tests

##### STEPS TO REPRODUCE

Run tests on Shippable.

##### EXPECTED RESULTS

Tests pass.

##### ACTUAL RESULTS

Tests are unstable or fail.

> test/integration/targets/azure_rm_containerinstance/tasks/main.yml:98 / [localhost] azure: azure_rm_containerinstance : Assert the container instance is deleted that=[u'output.changed']

> failure: rc=0

```
{
    "assertion": "output.changed", 
    "changed": false, 
    "evaluated_to": false, 
    "failed": true
}
```

> test/integration/targets/azure_rm_containerinstance/tasks/main.yml:1 / [localhost] azure: azure_rm_containerinstance : Create sample container instance name=aci{{ resource_group | hash('md5') | truncate(7, True, '') }}, resource_group={{ resource_group }}, os_type=linux, ip_address=public, ports=[80], containers=[{u'image': u'httpd', u'ports': [80, 81], u'name': u'mycontainer1', u'memory': 1.5}, {u'image': u'httpd', u'name': u'mycontainer2', u'memory': 1.5}], location=eastus

> error: Message: The container group 'aciaebe055' is still transitioning, please retry later.

```
Traceback (most recent call last):
  File "/tmp/ansible_1CMYew/ansible_module_azure_rm_containerinstance.py", line 424, in <module>
    main()
  File "/tmp/ansible_1CMYew/ansible_module_azure_rm_containerinstance.py", line 421, in main
    AzureRMContainerInstance()
  File "/tmp/ansible_1CMYew/ansible_module_azure_rm_containerinstance.py", line 264, in __init__
    supports_tags=False)
  File "/tmp/ansible_1CMYew/ansible_modlib.zip/ansible/module_utils/azure_rm_common.py", line 345, in __init__
  File "/tmp/ansible_1CMYew/ansible_module_azure_rm_containerinstance.py", line 318, in exec_module
    response = self.create_update_containerinstance()
  File "/tmp/ansible_1CMYew/ansible_module_azure_rm_containerinstance.py", line 384, in create_update_containerinstance
    container_group=parameters)
  File "/usr/local/lib/python2.7/dist-packages/azure/mgmt/containerinstance/operations/container_groups_operations.py", line 307, in create_or_update
    raise exp
msrestazure.azure_exceptions.CloudError: Azure Error: ContainerGroupTransitioning
Message: The container group 'aciaebe055' is still transitioning, please retry later.
```

**Comment by mattclay** [comment]
@zikalino Can you take a look at this?

**Comment by zikalino** [comment]
@mattclay i will investigate these issues. they don't seem to be related to Ansible in any way though...

**Comment by mattclay** [comment]
@zikalino At a minimum there is a bug in the module (or one of its dependencies), as modules should not traceback. Either the error should be handled internally within the module or it should be reported as an error without a traceback.

If the test is expected to fail intermittently due to errors reported by the module, then the test may not be suitable for running in CI.

**Comment by zikalino** [comment]
working on it

**Comment by mattclay** [comment]
Resolved by #40780
